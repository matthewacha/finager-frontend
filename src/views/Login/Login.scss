.login-background{
    height: 95%;
    width: 100%;
    display: inline-flex;
    @media screen and (min-width: 760px) and (max-width: 1024px) {
        display: block;
    }
    .auth{
        &__left-column{
            width: 40%;
            height: 100%;
            @media screen and (max-width: 550px){
                width: 100%;
            }
        }
        &__right-column{
            width: 60%;
            height: 100%;
            @media screen and (max-width: 550px){
                width: 0%;
                height: inherit;
                visibility: collapse;
            }
            &__panel{
                width: auto;
                height: 22.9%;
                background: #016682eb;
                text-align: center;

                &--text{
                    color: aliceblue;
                    font-family: 'Tahoma';
                    font-size: 20px;
                    display: flex;
                    height: 16.6%;
                }

                &-info{
                    height: 54.6%;
                    &_content{
                        height: inherit;
                        padding: 10%;
                        font-size: 2.5em;
                        text-transform: uppercase;
                        color: #8ca3aa;
                        font-weight: 600;
                        text-align: -webkit-center;
                        #examples{
                            font-size: .5em;
                            color: #24728c;
                            text-align: center;
                        }
                    }
                }
                &--foot{
                    width: auto;
                    height: 5.5%;
                    border: 1px solid #016682;
                    background: #016682;
                }

            }
        }

        &__panel{
            width: 100%;
            height: inherit;
            border-radius: 4px;
            background-color: #f0f8ff;
            &--icon{
                margin: auto;
                padding-top: 3%;
                width: 40%;
                height: 130%;
                border: 1px solid #97938f40;
                border-radius: 4px;
                position: relative;
                top: 35%;
                box-shadow: rgba(0, 0, 0, 0.2) 0px 1rem 1rem -1rem;
                z-index: 2;
                background: white;
                &__back{
                    position: relative;
                    z-index: 1;
                    padding: 10%;
                    width: auto;
                    height: 15%;
                    border: 1px solid #00000000;
                    background: #016682;
                }
                .icon-image{
                    width: -webkit-fill-available !important;
                    height: 90%!important;
                }
            }
            &--icon:hover {
                cursor: pointer;
            }

            &--text{
                padding: 10%;
                width: auto;
                height: 15%;
            }
            &--intermediate{
                width: auto;
                height: 55.3%;
                .app__title{
                    color: #016682eb;
                    position: relative;
                    left: 8px;
                    align-self: center;
                    text-transform: uppercase;
                    font-size: 2.5rem;
                    font-weight: 700;
                    &__border{
                        display: inline-flex;
                        position: relative;
                        left: 130px;
                        margin-left: 11%;
                        padding: 10px;
                        width: 22%;
                        height: 100%;
                        border-radius: 4px;
                    }
                }
            }
            &--social{
                display: inline-flex;
                margin-left: 21%;
                padding: 10px;
                width: 50%;
                height: 8%;
                border-radius: 10px;
                border: 2px solid #8ca3aa78;
                background-color: #f5fafa35;
            }
            &--social:hover {
                cursor: pointer;
                background-color: #f5fafa35;
                box-shadow: rgba(0, 0, 0, 0.1) 1px 1rem 1rem -1rem;
                transition: background-color 0.5s, box-shadow 0.2s ease-out;
            }
            &--video{
                display: inline-flex;
                position: relative;
                top: 30%;
                left: 25px;
                margin-left: 11%;
                padding: 10px;
                width: 26%;
                height: 30px;
                border-radius: 10px;
                border: 2px solid #8ca3aa;
            }
            &--video:hover {
                cursor: pointer;
                background-color: #f0f8ff;
                transition: background-color 0.2s ease-out;
            }
            
            &--try{
                display: inline-flex;
                position: relative;
                top: 30%;
                left: 25px;
                margin-left: 11%;
                padding: 10px;
                width: 26%;
                height: 30px;
                border-radius: 10px;
                border: 2px solid #8ca3aa;
            }

            &--try:hover {
                cursor: pointer;
                background-color: #f0f8ff;
                transition: background-color 0.2s ease-out;
            }

            &--terms{
                margin-left: 16%;
                font-size: 15px;
                height: 7%;
                padding-top: 6px;
            }
            &--terms:hover {
                cursor: pointer;
                color: #016682;
                text-decoration: underline solid #016682;
            }

            &--foot{
                width: auto;
                height: 5.5%;
                border: 1px solid #016682;
                background: #016682;
            }
        }
    }

}

.google-icon{
    width: fit-content;
    @media screen and (max-width: 550px){
        width: 34px;
    }
}

.video-icon{
    width: fit-content;
    position: relative;
    left: 25px;
    @media screen and (max-width: 550px){
        width: 34px;
    }
}

.auth-text{
    color: #737f82;
    font-family: 'Tahoma';
    font-size: 1.5em;
    align-self: center;
}
.auth-text:hover{
    color: #016682;
}

.side-button__text{
    color: #016682;
    font-family: 'Tahoma';
    font-size: 22px;
    position: relative;
    left: 40px;
    align-self: center;
    &--try{
        position: relative;
        left: 84px;
        align-self: center;
        color: #016682;
    }
}

.auth__footer{
    height: 5%;
    background: #016682;
}