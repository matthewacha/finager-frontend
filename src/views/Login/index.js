import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { jwt_decode } from 'jwt-decode';
import { Cookies } from 'js-cookie';
import "./Login.scss";

import googleIcon from "../../assets/images/googleIcon.svg";
import appIcon from "../../assets/images/appIcon.svg";
import videoIcon from "../../assets/images/video.svg";
import Button from "../../components/Button/Button";
import {
  loginUser,
  loginUserFailure
} from "../../redux/actionCreators/Authenticate";

import { env } from '../../config/env';
import { errorNotify, successNotify } from '../../Utils/notifies';


class Login extends Component {
  state = {
    status: "signup"
  };

  shouldComponentUpdate(nextProps, nextState){
    if(this.props.token !== nextProps.token){
      return true
    }
    if(this.props.error !== nextProps.error){
      return true
    }
    return false
  }

  componentDidUpdate(){
    const { token, error } = this.props;
    // if(token){
    //   const decodedUser = jwt_decode(token);
    //   Cookies.set('currentUser', decodedUser, { expires: 2 });
    // }
    // if(error){
    //   errorNotify(error.error)
    // }
  }

  handleLoad = () => {
    const params = env;
    try{
      window.gapi.load("auth2", () => {
        // initialize
        window.gapi.auth2.init(params).then(
          res => {
            // listen for sign in state changes
            // window.gapi.auth2.getAuthInstance().isSignedIn.listen(this.updateSigninStatus);
            // check if user is logged in
            if (res.isSignedIn && res.isSignedIn.get()) {
              // handle success
              this.handleSuccess(res.currentUser.get());
            } else {
              this.signInGoogle()
            }
          },
          // handle error action creator err =>
          error => {
            errorNotify('Not a valid origin')
            // errorCallback(error);
          }
        );
      });
    }catch(error){
      errorNotify('Please check your internet connection...')
    }
  };

  loadGoogleClient = () => {
    //   Load the google API client and auth2 library
    const script = document.getElementsByTagName('script')[1];
    script.onload = this.handleLoad();
  };

  componentDidMount() {
    this.loadGoogleClient(this.handleLoad());
  }

  handleSuccess = response => {
    const { loginUser } = this.props;
    const userProfile = response.getBasicProfile()
    const authResponse = response.getAuthResponse()
    response.googleId = userProfile.getId()
    response.tokenId = authResponse.id_token
    response.accessToken = authResponse.access_token
    loginUser(JSON.stringify(response))
  }

  signInGoogle = () => {
    let auth
    const { loginUserFailure } = this.props;
    try{
      auth = window.gapi.auth2.getAuthInstance();
      auth.signIn().then(
        res => {
          // handle success action creator here
          this.handleSuccess(res);
        }).catch(err => {
          // handle error action creator here
          loginUserFailure(err);
        })
    }catch(error){
      errorNotify('Please check your internet connection...')
    }
  }

  handleClick = event => {
    event.preventDefault();
    this.signInGoogle()
  };

  handleTerms = event => {
    /* istanbul ignore next */
    event.preventDefault();
    console.log("Redirect to terms...");
  };

  handleTryOut = event => {
    /* istanbul ignore next */
    event.preventDefault();
    console.log("Trying out app...");
  };

  handleVideo = event => {
    /* istanbul ignore next */
    event.preventDefault();
    console.log("Redirect to YouTube...");
  };

  render() {
    return (
      <Fragment>
      <div className={"login-background"}>
        <div className={"auth__left-column"}>
          <div type="presentation" className={"auth__panel"}>
            <div type="presentation" className={"auth__right-column__panel"}>
              <div type="presentation" className={"auth__panel--icon"}>
                <img
                  src={appIcon}
                  alt={"google-icon"}
                  className={"icon-image"}
                />
              </div>
            </div>
            <div type="presentation" className={"auth__panel--intermediate"}>
              <Button
                hasIcon={false}
                customButtonStyle={"app__title__border"}
                disable={"false"}
                text={"Finager"}
                textStyle={"app__title"}
                onClick={()=>{}}
                hasText={true}
              />
            </div>
            <Button
              hasIcon={true}
              iconAlt={"google-icon"}
              iconStyle={"google-icon"}
              iconImage={googleIcon}
              customButtonStyle={"auth__panel--social"}
              disable={"false"}
              text={"Login with Google"}
              textStyle={"auth-text"}
              onClick={this.handleClick}
              hasText={true}
            />
            <div
              role="button"
              onClick={this.handleTerms}
              className={"auth__panel--terms"}
            >
            Login confirms that you agree to our terms and
              conditions.
            </div>
          </div>
        </div>
        <div className={"auth__right-column"}>
          <div type="presentation" className={"auth__right-column__panel"} />
          <div className={"auth__right-column__panel-info"}>
            <div className={"auth__right-column__panel-info_content"}>
            <span>Automate and track all your routine financial transactions</span>
            <br></br>
            <span id={"examples"}>---Bills. loan payments. investments. donations---</span>
            </div>
          </div>
          <div className={"auth__right-column__panel--text"}>
            <Button
              hasIcon={true}
              iconAlt={"video"}
              iconStyle={"video-icon"}
              iconImage={videoIcon}
              customButtonStyle={"auth__panel--video"}
              disable={"false"}
              text={"Watch Tutorial"}
              textStyle={"side-button__text"}
              onClick={this.handleVideo}
              hasText={true}
            />
            <Button
              hasIcon={false}
              customButtonStyle={"auth__panel--try"}
              disable={"false"}
              text={"Demo"}
              textStyle={"side-button__text--try"}
              onClick={this.handleTryOut}
              hasText={true}
            />
          </div>
          </div>
      </div>
      <div type="presentation" className={"auth__footer"} />
      </Fragment>
    );
  }
}

Login.propTypes = {
  loginUser: PropTypes.func,
  loginUserFailure: PropTypes.func
};

Login.defaultProps = {
  loginUser: () => {},
  loginUserFailure: () => {}
};

const mapStateToProps = ({ auth }) => ({
  error: auth.error,
  token: auth.userToken
});

const actionCreators = {
  loginUserFailure,
  loginUser
};

export default connect(
  mapStateToProps,
  actionCreators
)(Login);
