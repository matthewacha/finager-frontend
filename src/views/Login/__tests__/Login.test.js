/* eslint-disable react/react-in-jsx-scope */
import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { notify } from 'react-notify-toast';

import Login from '../index';


Object.defineProperty(notify, 'show', {
    value: (...params) => 'show message'
})

Object.defineProperty(document, 'getElementsByTagName', {
    value: (...params) => ['', {onload: true}]
})

Object.defineProperty(window, 'gapi', {
    value: {auth2: {}}
})

Object.defineProperty(window.gapi, 'load', {
    value: (item, callBack) => {
        callBack()
    }
})

const props = {
    loginUser: jest.fn(),
}

const mockStore = configureStore();

const initialStore = {
    auth: {
        error: undefined,
    }

};

const store = mockStore(initialStore);

describe('Test Authenticate component', () => {
    it('Test it renders correctly', () => {
        const wrapper = shallow(
            <Provider store={store} >
                <Login store={store} props={props} />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    })
})

describe('Test renders correctly', () => {
    it('should load google api with signed in user', () => {
        Object.defineProperty(window.gapi.auth2, 'init', {
            value: (...params) => new Promise(resolve => resolve({
                    isSignedIn: { get: jest.fn(()=>true)},
                    currentUser:  {
                        get: () => {
                            return ({
                                getBasicProfile: jest.fn(() => {
                                    return({getId: () => '3445'})}),
                                getAuthResponse: jest.fn(() => {
                                    return({id_token: '1234', access_token: '4567'})}),})
                        }
                    }
                })),
                writable: true
        })
        const wrapper = mount(
            <Provider store={store}>
                <Login store={store} props={props}/>
            </Provider>)
        expect(wrapper).toMatchSnapshot();
    })

    describe('It renders correectly when user not signed in', () => {
        beforeAll(() => {
            Object.defineProperty(window.gapi.auth2, 'init', {
                value: (...params) => new Promise(resolve => resolve({
                        isSignedIn: { get: jest.fn(()=>false)},
                        currentUser:  {
                            get: () => {
                                return ({
                                    getBasicProfile: jest.fn(() => {
                                        return({getId: () => '3445'})}),
                                    getAuthResponse: jest.fn(() => {
                                        return({id_token: '1234', access_token: '4567'})}),})
                            }
                        }
                    }))
            })
        })

        it('should load google api with user not signed in', () => {
            Object.defineProperty(window.gapi.auth2, 'getAuthInstance', {
                value: () => {
                    return({ signIn : () =>new Promise(resolve => resolve(
                        {
                            getBasicProfile: jest.fn(() => {
                                return({getId: () => '3445'})}),
                            getAuthResponse: jest.fn(() => {
                                return({id_token: '1234', access_token: '4567'})}),}
                    ))})},
                    writable: true
            })
            const wrapper = mount(
                <Provider store={store}>
                    <Login store={store} props={props}/>
                </Provider>)
            expect(wrapper).toMatchSnapshot();
        })

        it('should load google api with user not signed in and fails to sign in', () => {
            Object.defineProperty(window.gapi.auth2.getAuthInstance, 'signIn', {
                value: () => new Promise((resolve, reject) => reject(new Error('Failed to get allocations'))),
                writable: true
            })

            Object.defineProperty(window.gapi.auth2, 'getAuthInstance', {
                value: () => {
                    return({ signIn : () => new Promise((resolve, reject) => reject(new Error('Failed to get allocations')))})},
                    writable: true
            })
            const wrapper = mount(
                <Provider store={store}>
                    <Login store={store} props={props}/>
                </Provider>)
            expect(wrapper).toMatchSnapshot();
        })

        it('should not sign in google api with user when unexpected error occurs', () => {
            Object.defineProperty(window.gapi.auth2, 'getAuthInstance', {
                value: () => new Error('Failed to get allocations'),
                    writable: true
            })
            const wrapper = mount(
                <Provider store={store}>
                    <Login store={store} props={props}/>
                </Provider>)
            expect(wrapper).toMatchSnapshot();
        })

        it('should catch error when offline', () => {
            Object.defineProperty(window.gapi.auth2, 'init', {
                value: (...params) => new Error('Invalid'),
                    writable: true
            })
            const wrapper = mount(
                <Provider store={store}>
                    <Login store={store} props={props}/>
                </Provider>)
            expect(wrapper).toMatchSnapshot();
        })

        it('should not load from invalid origin', () => {
            Object.defineProperty(window.gapi.auth2, 'init', {
                value: (...params) => new Promise((resolve, reject) => reject(new Error('Invalid origin')))
            })
            const wrapper = mount(
                <Provider store={store}>
                    <Login store={store} props={props}/>
                </Provider>)
            expect(wrapper).toMatchSnapshot();
        })

    })

})

describe('Test Login click', () => {
    it('should signin on click', () => {
        const wrapper = mount(
            <Provider store={store}>
                <Login store={store} props={props}/>
            </Provider>)

        const componentInstance = wrapper.childAt(0).childAt(0).instance()
        const spyThis = jest.spyOn(componentInstance, 'signInGoogle')
        const loginButton = wrapper.find('.auth__panel--social')
        loginButton.simulate('click', { preventDefault: () => {} })
        expect(spyThis).toHaveBeenCalled()
    })
})