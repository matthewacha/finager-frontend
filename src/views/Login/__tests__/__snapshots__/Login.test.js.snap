// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Authenticate component Test it renders correctly 1`] = `
<Connect(Login)
  props={
    Object {
      "loginUser": [MockFunction],
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`Test renders correctly It renders correectly when user not signed in should catch error when offline 1`] = `
Array [
  <div
    className="login-background"
  >
    <div
      className="auth__left-column"
    >
      <div
        className="auth__panel"
        type="presentation"
      >
        <div
          className="auth__right-column__panel"
          type="presentation"
        >
          <div
            className="auth__panel--icon"
            type="presentation"
          >
            <img
              alt="google-icon"
              className="icon-image"
              src="appIcon.svg"
            />
          </div>
        </div>
        <div
          className="auth__panel--intermediate"
          type="presentation"
        >
          <div
            className="app__title__border"
            disable="false"
            onClick={[Function]}
            role="button"
          >
            <span
              className="app__title"
            >
              Finager
            </span>
          </div>
        </div>
        <div
          className="auth__panel--social"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="google-icon"
            className="google-icon"
            src="googleIcon.svg"
          />
          <span
            className="auth-text"
          >
            Login with Google
          </span>
        </div>
        <div
          className="auth__panel--terms"
          onClick={[Function]}
          role="button"
        >
          Login confirms that you agree to our terms and conditions.
        </div>
      </div>
    </div>
    <div
      className="auth__right-column"
    >
      <div
        className="auth__right-column__panel"
        type="presentation"
      />
      <div
        className="auth__right-column__panel-info"
      >
        <div
          className="auth__right-column__panel-info_content"
        >
          <span>
            Automate and track all your routine financial transactions
          </span>
          <br />
          <span
            id="examples"
          >
            ---Bills. loan payments. investments. donations---
          </span>
        </div>
      </div>
      <div
        className="auth__right-column__panel--text"
      >
        <div
          className="auth__panel--video"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="video"
            className="video-icon"
            src="video.svg"
          />
          <span
            className="side-button__text"
          >
            Watch Tutorial
          </span>
        </div>
        <div
          className="auth__panel--try"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <span
            className="side-button__text--try"
          >
            Demo
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="auth__footer"
    type="presentation"
  />,
]
`;

exports[`Test renders correctly It renders correectly when user not signed in should load google api with user not signed in 1`] = `
Array [
  <div
    className="login-background"
  >
    <div
      className="auth__left-column"
    >
      <div
        className="auth__panel"
        type="presentation"
      >
        <div
          className="auth__right-column__panel"
          type="presentation"
        >
          <div
            className="auth__panel--icon"
            type="presentation"
          >
            <img
              alt="google-icon"
              className="icon-image"
              src="appIcon.svg"
            />
          </div>
        </div>
        <div
          className="auth__panel--intermediate"
          type="presentation"
        >
          <div
            className="app__title__border"
            disable="false"
            onClick={[Function]}
            role="button"
          >
            <span
              className="app__title"
            >
              Finager
            </span>
          </div>
        </div>
        <div
          className="auth__panel--social"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="google-icon"
            className="google-icon"
            src="googleIcon.svg"
          />
          <span
            className="auth-text"
          >
            Login with Google
          </span>
        </div>
        <div
          className="auth__panel--terms"
          onClick={[Function]}
          role="button"
        >
          Login confirms that you agree to our terms and conditions.
        </div>
      </div>
    </div>
    <div
      className="auth__right-column"
    >
      <div
        className="auth__right-column__panel"
        type="presentation"
      />
      <div
        className="auth__right-column__panel-info"
      >
        <div
          className="auth__right-column__panel-info_content"
        >
          <span>
            Automate and track all your routine financial transactions
          </span>
          <br />
          <span
            id="examples"
          >
            ---Bills. loan payments. investments. donations---
          </span>
        </div>
      </div>
      <div
        className="auth__right-column__panel--text"
      >
        <div
          className="auth__panel--video"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="video"
            className="video-icon"
            src="video.svg"
          />
          <span
            className="side-button__text"
          >
            Watch Tutorial
          </span>
        </div>
        <div
          className="auth__panel--try"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <span
            className="side-button__text--try"
          >
            Demo
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="auth__footer"
    type="presentation"
  />,
]
`;

exports[`Test renders correctly It renders correectly when user not signed in should load google api with user not signed in and fails to sign in 1`] = `
Array [
  <div
    className="login-background"
  >
    <div
      className="auth__left-column"
    >
      <div
        className="auth__panel"
        type="presentation"
      >
        <div
          className="auth__right-column__panel"
          type="presentation"
        >
          <div
            className="auth__panel--icon"
            type="presentation"
          >
            <img
              alt="google-icon"
              className="icon-image"
              src="appIcon.svg"
            />
          </div>
        </div>
        <div
          className="auth__panel--intermediate"
          type="presentation"
        >
          <div
            className="app__title__border"
            disable="false"
            onClick={[Function]}
            role="button"
          >
            <span
              className="app__title"
            >
              Finager
            </span>
          </div>
        </div>
        <div
          className="auth__panel--social"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="google-icon"
            className="google-icon"
            src="googleIcon.svg"
          />
          <span
            className="auth-text"
          >
            Login with Google
          </span>
        </div>
        <div
          className="auth__panel--terms"
          onClick={[Function]}
          role="button"
        >
          Login confirms that you agree to our terms and conditions.
        </div>
      </div>
    </div>
    <div
      className="auth__right-column"
    >
      <div
        className="auth__right-column__panel"
        type="presentation"
      />
      <div
        className="auth__right-column__panel-info"
      >
        <div
          className="auth__right-column__panel-info_content"
        >
          <span>
            Automate and track all your routine financial transactions
          </span>
          <br />
          <span
            id="examples"
          >
            ---Bills. loan payments. investments. donations---
          </span>
        </div>
      </div>
      <div
        className="auth__right-column__panel--text"
      >
        <div
          className="auth__panel--video"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="video"
            className="video-icon"
            src="video.svg"
          />
          <span
            className="side-button__text"
          >
            Watch Tutorial
          </span>
        </div>
        <div
          className="auth__panel--try"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <span
            className="side-button__text--try"
          >
            Demo
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="auth__footer"
    type="presentation"
  />,
]
`;

exports[`Test renders correctly It renders correectly when user not signed in should not load from invalid origin 1`] = `
Array [
  <div
    className="login-background"
  >
    <div
      className="auth__left-column"
    >
      <div
        className="auth__panel"
        type="presentation"
      >
        <div
          className="auth__right-column__panel"
          type="presentation"
        >
          <div
            className="auth__panel--icon"
            type="presentation"
          >
            <img
              alt="google-icon"
              className="icon-image"
              src="appIcon.svg"
            />
          </div>
        </div>
        <div
          className="auth__panel--intermediate"
          type="presentation"
        >
          <div
            className="app__title__border"
            disable="false"
            onClick={[Function]}
            role="button"
          >
            <span
              className="app__title"
            >
              Finager
            </span>
          </div>
        </div>
        <div
          className="auth__panel--social"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="google-icon"
            className="google-icon"
            src="googleIcon.svg"
          />
          <span
            className="auth-text"
          >
            Login with Google
          </span>
        </div>
        <div
          className="auth__panel--terms"
          onClick={[Function]}
          role="button"
        >
          Login confirms that you agree to our terms and conditions.
        </div>
      </div>
    </div>
    <div
      className="auth__right-column"
    >
      <div
        className="auth__right-column__panel"
        type="presentation"
      />
      <div
        className="auth__right-column__panel-info"
      >
        <div
          className="auth__right-column__panel-info_content"
        >
          <span>
            Automate and track all your routine financial transactions
          </span>
          <br />
          <span
            id="examples"
          >
            ---Bills. loan payments. investments. donations---
          </span>
        </div>
      </div>
      <div
        className="auth__right-column__panel--text"
      >
        <div
          className="auth__panel--video"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="video"
            className="video-icon"
            src="video.svg"
          />
          <span
            className="side-button__text"
          >
            Watch Tutorial
          </span>
        </div>
        <div
          className="auth__panel--try"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <span
            className="side-button__text--try"
          >
            Demo
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="auth__footer"
    type="presentation"
  />,
]
`;

exports[`Test renders correctly It renders correectly when user not signed in should not sign in google api with user when unexpected error occurs 1`] = `
Array [
  <div
    className="login-background"
  >
    <div
      className="auth__left-column"
    >
      <div
        className="auth__panel"
        type="presentation"
      >
        <div
          className="auth__right-column__panel"
          type="presentation"
        >
          <div
            className="auth__panel--icon"
            type="presentation"
          >
            <img
              alt="google-icon"
              className="icon-image"
              src="appIcon.svg"
            />
          </div>
        </div>
        <div
          className="auth__panel--intermediate"
          type="presentation"
        >
          <div
            className="app__title__border"
            disable="false"
            onClick={[Function]}
            role="button"
          >
            <span
              className="app__title"
            >
              Finager
            </span>
          </div>
        </div>
        <div
          className="auth__panel--social"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="google-icon"
            className="google-icon"
            src="googleIcon.svg"
          />
          <span
            className="auth-text"
          >
            Login with Google
          </span>
        </div>
        <div
          className="auth__panel--terms"
          onClick={[Function]}
          role="button"
        >
          Login confirms that you agree to our terms and conditions.
        </div>
      </div>
    </div>
    <div
      className="auth__right-column"
    >
      <div
        className="auth__right-column__panel"
        type="presentation"
      />
      <div
        className="auth__right-column__panel-info"
      >
        <div
          className="auth__right-column__panel-info_content"
        >
          <span>
            Automate and track all your routine financial transactions
          </span>
          <br />
          <span
            id="examples"
          >
            ---Bills. loan payments. investments. donations---
          </span>
        </div>
      </div>
      <div
        className="auth__right-column__panel--text"
      >
        <div
          className="auth__panel--video"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="video"
            className="video-icon"
            src="video.svg"
          />
          <span
            className="side-button__text"
          >
            Watch Tutorial
          </span>
        </div>
        <div
          className="auth__panel--try"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <span
            className="side-button__text--try"
          >
            Demo
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="auth__footer"
    type="presentation"
  />,
]
`;

exports[`Test renders correctly should load google api with signed in user 1`] = `
Array [
  <div
    className="login-background"
  >
    <div
      className="auth__left-column"
    >
      <div
        className="auth__panel"
        type="presentation"
      >
        <div
          className="auth__right-column__panel"
          type="presentation"
        >
          <div
            className="auth__panel--icon"
            type="presentation"
          >
            <img
              alt="google-icon"
              className="icon-image"
              src="appIcon.svg"
            />
          </div>
        </div>
        <div
          className="auth__panel--intermediate"
          type="presentation"
        >
          <div
            className="app__title__border"
            disable="false"
            onClick={[Function]}
            role="button"
          >
            <span
              className="app__title"
            >
              Finager
            </span>
          </div>
        </div>
        <div
          className="auth__panel--social"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="google-icon"
            className="google-icon"
            src="googleIcon.svg"
          />
          <span
            className="auth-text"
          >
            Login with Google
          </span>
        </div>
        <div
          className="auth__panel--terms"
          onClick={[Function]}
          role="button"
        >
          Login confirms that you agree to our terms and conditions.
        </div>
      </div>
    </div>
    <div
      className="auth__right-column"
    >
      <div
        className="auth__right-column__panel"
        type="presentation"
      />
      <div
        className="auth__right-column__panel-info"
      >
        <div
          className="auth__right-column__panel-info_content"
        >
          <span>
            Automate and track all your routine financial transactions
          </span>
          <br />
          <span
            id="examples"
          >
            ---Bills. loan payments. investments. donations---
          </span>
        </div>
      </div>
      <div
        className="auth__right-column__panel--text"
      >
        <div
          className="auth__panel--video"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <img
            alt="video"
            className="video-icon"
            src="video.svg"
          />
          <span
            className="side-button__text"
          >
            Watch Tutorial
          </span>
        </div>
        <div
          className="auth__panel--try"
          disable="false"
          onClick={[Function]}
          role="button"
        >
          <span
            className="side-button__text--try"
          >
            Demo
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="auth__footer"
    type="presentation"
  />,
]
`;
